crates/packager/src/package/pacman/mod.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024-2024 CrabNebula Ltd.
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use super::{
+    deb::{copy_custom_files, generate_data, tar_and_gzip_dir},
+    Context,
+};
+use crate::{config::Config, util};
+use heck::AsKebabCase;
+use sha2::{Digest, Sha512};
+use std::{
+    fs::File,
+    io::{self, Write},
+    path::{Path, PathBuf},
+};
+
+#[tracing::instrument(level = "trace")]
+pub(crate) fn package(ctx: &Context) -> crate::Result<Vec<PathBuf>> {
+    let Context {
+        config,
+        intermediates_path,
+        ..
+    } = ctx;
+
+    let arch = match config.target_arch()? {
+        "x86" => "i386",
+        "arm" => "armhf",
+        other => other,
+    };
+
+    let intermediates_path = intermediates_path.join("pacman");
+    util::create_clean_dir(&intermediates_path)?;
+
+    let package_base_name = format!("{}_{}_{}", config.main_binary_name()?, config.version, arch);
+    let package_name = format!("{}.tar.gz", package_base_name);
+
+    let pkg_dir = intermediates_path.join(&package_base_name);
+    let pkg_path = config.out_dir().join(&package_name);
+    let pkgbuild_path = pkg_path.with_file_name("PKGBUILD");
+
+    tracing::info!("Packaging {} ({})", package_name, pkg_path.display());
+
+    tracing::debug!("Generating data");
+    let _ = generate_data(config, &pkg_dir)?;
+
+    tracing::debug!("Copying files specified in `deb.files`");
+    if let Some(files) = config.deb().and_then(|d| d.files.as_ref()) {
+        copy_custom_files(files, &pkg_dir)?;
+    }
+
+    // Apply tar/gzip to create the final package file.
+    tracing::debug!("Creating package archive using tar and gzip");
+    let data_tar_gz_path = tar_and_gzip_dir(pkg_dir)?;
+    std::fs::copy(data_tar_gz_path, &pkg_path)?;
+
+    tracing::info!("Generating PKGBUILD: {}", pkgbuild_path.display());
+    generate_pkgbuild_file(config, arch, pkgbuild_path.as_path(), pkg_path.as_path())?;
+
+    Ok(vec![pkg_path])
+}
+
+/// Generates the pacman PKGBUILD file.
+/// For more information about the format of this file, see
+/// <https://wiki.archlinux.org/title/PKGBUILD>
+fn generate_pkgbuild_file(
+    config: &Config,
+    arch: &str,
+    dest_dir: &Path,
+    package_path: &Path,
+) -> crate::Result<()> {
+    let pkgbuild_path = dest_dir.with_file_name("PKGBUILD");
+    let mut file = util::create_file(&pkgbuild_path)?;
+
+    if let Some(authors) = &config.authors {
+        writeln!(file, "# Maintainer: {}", authors.join(", "))?;
+    }
+    writeln!(file, "pkgname={}-bin", AsKebabCase(&config.product_name))?;
```

Comment:
why do we need to make it kebab case? and also why do we add `-bin` suffix?

Reply:
Kebab case is mandatory I guess, but is surely recommended if not. 
`-bin` suggests that it is a prebuilt package, and we don't have to compile it.

Reply:
> We do use it for deb as well, see `fn generate_control_file`

I see, 

> `-bin` prefix will be shown at `PKGBUILD` , and hence AUR. For reference, see how different versions of VS Code is named in Arch Linux
> 
> [wiki.archlinux.org/title/Visual_Studio_Code](https://wiki.archlinux.org/title/Visual_Studio_Code)
> 
> Pre built binary have `bin` as suffix in AUR, which is recommended.
> 
> [aur.archlinux.org/packages/visual-studio-code-bin](https://aur.archlinux.org/packages/visual-studio-code-bin)

my main concern is that users will want their package to be installed using `pacman -S tauri-app` not `pacman -S tauri-app-bin`, is this still the case or adding `-bin` will change the install command as well? I am fine with `-bin` for AUR

Reply:
Yes, it will also change the name for pacman command, but to be honest, I don't think that would matter.

People just search the AUR everytime they want to install it, because app names are confusing most of the times. 

Also, this thread might be a useful refernce

https://forum.manjaro.org/t/in-aur-what-is-the-difference-between-package-package-bin-and-package-git/149697

We can certainly remove that if you want it that way.

Reply:
Oh, That's cool you use Arch as well :grin: 

Certainly, we can remove `bin` in that case.